WorkloadManagementSystem/PilotAgent/PilotLogger/PilotLoggerTools.py

```diff
@@ -0,0 +1,220 @@
+      key_file,
+      cert_file,
+      ca_certs,
+      fileWithID
+      ]
+  config = dict( zip( keys, values ) )
+  config = json.dumps(config)
+  with open(filename, 'w') as myFile:
+    myFile.write(config)
+
+def readPilotLoggerConfigFile ( filename ):
+  """Helper function that loads configuration file.
+  Returns:
+    dict:
+  """
+  try:
+    with open(filename, 'r') as myFile:
+      config = myFile.read()
+      config = json.loads(config)
+      return config
+  except (IOError, ValueError):
+    return None
+
+def generateDict( pilotUUID, status, minorStatus, timestamp, source ):
+  """Helper function that returs a dictionnary based on the
+     set of input values.
+  Returns
+    dict:
+  """
+
+  keys = [
+      'pilotUUID',
+      'status',
+      'minorStatus',
+      'timestamp',
+      'source'
+      ]
+  values = [
+      pilotUUID,
+      status,
+      minorStatus,
+      timestamp,
+      source
+      ]
+  return dict( zip( keys, values ) )
+
+def encodeMessage( content ):
+  """Method encodes the message in form of the serialized JSON string
+     see https://docs.python.org/2/library/json.html#py-to-json-table
+  Args:
+    content(dict):
+  Returns:
+    str: in the JSON format.
+  Raises:
+    TypeError:if cannont encode json properly
+  """
+  return json.dumps( content )
+
+def decodeMessage( msgJSON ):
+  """Decodes the message from the serialized JSON string
+     See https://docs.python.org/2/library/json.html#py-to-json-table.
+  Args:
+    msgJSON(str):in the JSON format.
+  Returns:
+    str: decoded objecst.
+  Raises:
+    TypeError: if cannot decode JSON properly.
+  """
+  return json.loads( msgJSON )
+
+def isMessageFormatCorrect( content ):
+  """Checks if input format is correct.
+     Function checks if the input format is a dictionnary
+     in the following format:
+     0) content is a dictionary,
+     1) it contains only those keys of basestring types:
+     'pilotUUID', 'status', 'minorStatus', 'timestamp', 'source',
+     2) it contains only values of basestring types.
+  Args:
+    content(dict): all values must be non-empty
+  Returns:
+    bool: True if message format is correct, False otherwise
+  Example:
+    {"status": "DIRAC Installation",
+     "timestamp": "1427121370.7",
+      "minorStatus": "Uname = Linux localhost 3.10.64-85.cernvm.x86_64",
+      "pilotUUID": "eda78924-d169-11e4-bfd2-0800275d1a0a",
+      "source": "pilot"
+      }
+  """
+  if not isinstance( content, dict ):
+    return False
+  refKeys = [
+      'pilotUUID',
+      'status',
+      'minorStatus',
+      'timestamp',
+      'source'
+      ]
+  refKeys.sort()
+  keys = content.keys()
+  keys.sort()
+  if not keys == refKeys:
+    return False
+  values = content.values()
+  # if any value is not of basestring type
+  if any( not isinstance( val, basestring ) for val in values ):
+    return False
+  #checking if not empty for all
+  values = content.values()
+  if any( not val for val in values ):
```

Comment:
Your comment says "checking if not empty for all", so that means that they should not all be empty, right ? Because this check is True if at least one of the value is None, so it is not the same.